pdata/internal/cmd/pdatagen/internal/base_fields.go

```diff
@@ -16,272 +16,284 @@ package internal // import "go.opentelemetry.io/collector/pdata/internal/cmd/pda
 
 import (
 	"bytes"
-	"os"
 	"strings"
+	"text/template"
 )
 
-const accessorSliceTemplate = `// ${fieldName} returns the ${fieldName} associated with this ${structName}.
-func (ms ${structName}) ${fieldName}() ${returnType} {
-	return new${returnType}(&ms.${origAccessor}.${fieldName})
+const accessorSliceTemplate = `// {{ .fieldName }} returns the {{ .fieldName }} associated with this {{ .structName }}.
+func (ms {{ .structName }}) {{ .fieldName }}() {{ .packageName }}{{ .returnType }} {
+	{{- if .isCommon }}
+	return {{ .packageName }}{{ .returnType }}(internal.New{{ .returnType }}(&ms.{{ .origAccessor }}.{{ .fieldName }}))
+	{{- else }}
+	return new{{ .returnType }}(&ms.{{ .origAccessor }}.{{ .fieldName }})
+	{{- end }}
 }`
 
-const accessorSliceTemplateCommon = `// ${fieldName} returns the ${fieldName} associated with this ${structName}.
-func (ms ${structName}) ${fieldName}() ${packageName}${returnType} {
-	return ${packageName}${returnType}(internal.New${returnType}(&ms.${origAccessor}.${fieldName}))
```

Comment:
If I understand correctly, we're now autodetecting whether or not each struct is common based on the package, and this is necessarily part of this change in order to build the map expected by go templates. Is that right?

Reply:
> now autodetecting whether or not each struct is common based on the package

We've been doing that for a while already. But it's still can be changed if even needed